### PR TITLE
Log-Re-Conditioned Panel Cp: physics-adaptive inviscid scaling

### PR DESCRIPTION
**Pull request by tcapelle**
## Hypothesis

The Panel Cp feature (PR #2319) improved p_tan -1.7% and p_oodc -1.3% but caused **p_re +1.0% regression**. The flat-plate Cp (Cp = -2sin(α)/√(x(1-x))) ignores Reynolds number effects entirely. At extreme Re, the boundary layer physics change qualitatively — at high Re the inviscid approximation is more accurate (thinner BL), while at low Re viscous effects dominate and Cp deviates more from inviscid theory.

A **learned Re-conditioned scaling** allows the model to downweight the physics hint at low Re (where it's least accurate) and upweight it at high Re. This directly addresses the p_re regression.

**Reference:** Viscous-inviscid interaction theory (Drela, "XFOIL", 1989) documents how Re changes the departure of real Cp from inviscid theory. FiLM conditioning (Perez et al. 2018) is well-validated for parameter-adaptive feature scaling.

### Changes to `cfd_tandemfoil/train.py`

1. **Add flag:**
   ```python
   parser.add_argument('--cp_panel_re_cond', action='store_true', help='Re-conditioned panel Cp scaling')
   ```

2. **Implement a tiny Re→scale network:**
   ```python
   if args.cp_panel_re_cond:
       # Tiny MLP: log(Re) → scale in (0, 1)
       self.re_cp_scale = nn.Sequential(
           nn.Linear(1, 16),
           nn.GELU(),
           nn.Linear(16, 1),
           nn.Sigmoid()
       )
       # Initialize output bias so sigmoid(bias) ≈ 1.0 (identity at init)
       # sigmoid(4.6) ≈ 0.99, so init bias to ~4.6
       nn.init.constant_(self.re_cp_scale[-2].bias, 4.6)
   ```

3. **Apply in the feature construction**, where `cp_panel_scale` is multiplied:
   ```python
   if args.cp_panel_re_cond:
       # log_re: [B, 1] — extract from the condition features
       # Normalize: log(Re) is typically in [11, 16] range → normalize to [0, 1]
       log_re_norm = (log_re - 11.0) / 5.0  # approximate normalization
       re_scale = self.re_cp_scale(log_re_norm)  # [B, 1]
       # Apply per-sample: multiply the cp_panel feature by re_scale
       cp_features = cp_features * re_scale.unsqueeze(-1) * args.cp_panel_scale
   else:
       cp_features = cp_features * args.cp_panel_scale
   ```

4. **Important details:**
   - The Re→scale network is jointly trained with the rest of the model
   - Initialize to near-identity (sigmoid output ≈ 1.0) so it starts as the current baseline
   - The network learns to REDUCE the Cp scale at extreme Re values where inviscid theory is less accurate
   - Log `train/re_cp_scale_min` and `train/re_cp_scale_max` to W&B to monitor the learned Re→scale mapping

5. **Run 2 seeds:**
   ```bash
   CUDA_VISIBLE_DEVICES=0 python train.py \
     --agent thorfinn --wandb_name "thorfinn/re-cond-cp-s42" --wandb_group log-re-conditioned-cp \
     --seed 42 --asinh_pressure --field_decoder --adaln_output --use_lion --lr 2e-4 \
     --slice_num 96 --cosine_T_max 150 --pcgrad_3way \
     --pressure_first --pressure_deep --residual_prediction --surface_refine \
     --te_coord_frame --wake_deficit_feature --re_stratified_sampling --n_layers 3 \
     --cp_panel --cp_panel_tandem_only --cp_panel_scale 0.1 \
     --cp_panel_re_cond

   CUDA_VISIBLE_DEVICES=1 python train.py \
     --agent thorfinn --wandb_name "thorfinn/re-cond-cp-s73" --wandb_group log-re-conditioned-cp \
     --seed 73 [same flags as above]
   ```

## Baseline
| Metric | 2-seed avg | Target |
|--------|-----------|--------|
| **p_in** | **11.709** | < 11.71 |
| **p_oodc** | **7.544** | < 7.54 |
| **p_tan** | **27.402** | < 27.40 |
| p_re | 6.481 | < 6.48 |

W&B baseline: h6fqcry4 (s42), cuhoscp9 (s73)
Reproduce: `cd cfd_tandemfoil && python train.py --asinh_pressure --field_decoder --adaln_output --use_lion --lr 2e-4 --slice_num 96 --cosine_T_max 150 --pcgrad_3way --pressure_first --pressure_deep --residual_prediction --surface_refine --te_coord_frame --wake_deficit_feature --re_stratified_sampling --n_layers 3 --cp_panel --cp_panel_tandem_only --cp_panel_scale 0.1`

### CODE REVIEW

**Comment by tcapelle** [comment]
ADVISOR: Closing PR #2351 because ADVISOR: Closing after reviewing W&B runs.

**Results (2-seed avg vs baseline p_in=11.872, p_oodc=7.459, p_tan=26.319, p_re=6.229):**
| Seed | p_in | p_oodc | p_tan | p_re |
|------|------|--------|-------|------|
| s42 (joyoucec) | 11.755 ✅ | 7.511 ❌ | 26.208 ✅ | 6.336 ❌ |
| s73 (r0zyimb1) | 12.398 ❌ | 7.314 ✅ | 26.686 ❌ | 6.319 ❌ |
| **2-seed avg** | **12.077 ❌ (+1.7%)** | **7.413 ✅ (-0.6%)** | **26.447 ❌ (+0.5%)** | **6.328 ❌ (+1.6%)** |

The 2-seed average only beats baseline on p_oodc (-0.6%). p_in, p_tan, and p_re all regress. High seed variance (s42 p_in=11.755 vs s73 p_in=12.398, spread=0.643) indicates the Re-conditioning is not providing stable signal — the tiny Re→scale MLP is learning to approximate the identity at most Re values, not a meaningful physics-grounded correction.

The underlying intuition was sound (Re conditions the accuracy of inviscid theory), but the signal is too weak to overcome the noise at 2-seed evidence level. The physics-feature era is exhausted for now. Good experiment — time to move to bolder architectual changes.

You will get a bold Round 41 assignment shortly..
